vermouth/processors/apply_rubber_band.py

```diff
@@ -158,30 +163,24 @@ def build_connectivity_matrix(graph, separation, selected_nodes):
     numpy.ndarray
         A boolean matrix.
     """
-    if separation < 0:
-        raise ValueError('Separation has to be zero or positive.')
-    if separation == 0:
-        # The connectivity matrix with a separation of 0 is the adjacency
-        # matrix. Thankfully, networkx can directly give it to us a a numpy
-        # matrix.
-        return np.asarray(nx.to_numpy_matrix(graph, nodelist=selected_nodes).astype(bool))
+    res_graph = make_residue_graph(graph)
+    distance_pairs = nx.all_pairs_shortest_path_length(res_graph, cutoff=separation)
```

Comment:
What happens when make_residue_graph starts returning a DiGraph? I think this will still work, but the resulting matrix will simply become asymmetric.

Reply:
@pckroon well yes if the DiGraph is used you will get an asymmetric matrix and it will break probably 50% of the time. Is a directed residue graph sensible though?

Reply:
The only thing that would have to be changed then is the use of np.triu_indices, so that wouldn't be too bad.
And yes, very much. The most thorough discussion on it is probably in #57

Reply:
@pckroon so we put a comment with it, first? Using the full matrix will probably slow it down a lot for large protein complexes.

Reply:
Leave it as is for now. The general structure still works, it'll just be a fast bughunt